src/test/java/eu/dissco/core/translator/service/DwcaServiceTest.java

```diff
@@ -126,16 +175,32 @@ void testRetrieveDataWithAcMedia() throws IOException {
   void testRetrieveDataWithAssociatedMedia() throws IOException {
     // Given
     givenDWCA("/dwca-lux-associated-media.zip");
+    setupTermMapper();
+    given(dwcaRepository.getCoreRecords(anyList(), anyString())).willReturn(givenSpecimenMapWithMedia(20));
 
     // When
     service.retrieveData();
 
     // Then
+    then(dwcaRepository).should(times(1)).createTable(anyString());
+    then(dwcaRepository).should(times(1)).postRecords(anyString(), anyList());
     then(kafkaService).should(times(20)).sendMessage(eq("digital-specimen"), anyString());
-    then(kafkaService).should(times(21)).sendMessage(eq("digital-media-object"), anyString());
+    then(kafkaService).should(times(40)).sendMessage(eq("digital-media-object"), anyString());
```

Comment:
The amount of times you send the request, is it to test if the system is capable of taking all those requests at once?

Reply:
Not really to test the system but that it creates the correct amount of batches and calls.
In this case it should send 40 times a digital-media-object so if it suddenly fails then we broke something.